include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -1178,21 +1198,31 @@ def SliceEncodingAttr : DistributedEncoding<"SliceEncoding", "slice_encoding"> {
   let mnemonic = "slice";
 
   let description = [{
-    TODO: improve docs
+    Given a `parent` layout and a `dim`, squeezes the given `dim` in the `parent`
+    layout and distributes values in a tensor T according to the new layout.
+
+    For example, given
+
+    T = [x  x  x  x  x  x  x  x]
+    L_parent = [0  1  2  3 ]
+               [4  5  6  7 ]
+               [8  9  10 11]
+               [12 13 14 15] (with 16 CUDA threads)
 
-    A = [x  x  x  x  x  x  x  x]
+    With dim = 0, squeezing out dim 0, we have
+    L = [{0,4,8,12},  {1,5,9,13}, {2,6,10,14},  {3,7,11,15} ]
 
-    parent = [0  1  2  3 ]
-             [4  5  6  7 ]
-             [8  9  10 11]
-             [12 13 14 15]
-    dim = 0
+    Then the data of T would be distributed as follow between the 16 CUDA threads:
+    L(T) = [ {0,4,8,12} , {1,5,9,13} , ... {3,7,11,15}, {0,4,8,12} , ..., {3,7,11,15} ]
 
-    Then the data of A would be distributed as follow between the 16 CUDA threads:
-    L(A) = [ {0,4,8,12} , {1,5,9,13} , ... {3,7,11,15}, {0,4,8,12} , ..., {3,7,11,15} ]
+    With dim = 1, squeezing out dim 1, we have
```

Comment:
With dim = 1, it's very confusing to see how the mapping works with a row vector T. Maybe it's clearer with a column vector?

Reply:
Yup make sense changed.